usage/usage.go

```diff
@@ -0,0 +1,132 @@
+	client, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new pubsub client: %w", err)
+	}
+	return &Reporter{pubsubClient: client}, nil
+}
+
+// Close closes a Reporter and the underlying PubSub client.
+func (r *Reporter) Close() error {
+	return r.pubsubClient.Close()
+}
+
+// publishEvent uses the underlying PubSub client to publish
+// a marshaled KNEEvent proto message to the event topic.
+func (r *Reporter) publishEvent(ctx context.Context, event *epb.KNEEvent) error {
+	msg, err := proto.Marshal(event)
+	if err != nil {
+		return err
+	}
+	t := r.pubsubClient.Topic(topicID)
+	res := t.Publish(ctx, &pubsub.Message{Data: msg})
+	id, err := res.Get(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to publish message: %w", err)
+	}
+	log.V(2).Infof("Published event %s with message ID: %s", event.GetUuid(), id)
+	return nil
+}
+
+// ReportDeployClusterStart reports that a cluster deployment has started. The UUID of the
+// reported event is returned along with an error if the event reporting failed.
+func (r *Reporter) ReportDeployClusterStart(ctx context.Context, c *epb.Cluster) (string, error) {
+	event := newEvent()
+	event.Event = &epb.KNEEvent_DeployClusterStart{DeployClusterStart: &epb.DeployClusterStart{Cluster: c}}
+	if err := r.publishEvent(ctx, event); err != nil {
+		return "", fmt.Errorf("failed to report DeployClusterStart event with ID %s: %w", event.GetUuid(), err)
+	}
+	log.V(1).Infof("Reported DeployClusterStart event with ID %s", event.GetUuid())
+	return event.GetUuid(), nil
+}
+
+// ReportDeployClusterEnd reports that a cluster deployment has ended.
+func (r *Reporter) ReportDeployClusterEnd(ctx context.Context, id string, deployErr error) error {
+	event := newEvent()
+	event.Uuid = id
+	end := &epb.DeployClusterEnd{}
+	if deployErr != nil {
+		end.Error = deployErr.Error()
+	}
+	event.Event = &epb.KNEEvent_DeployClusterEnd{DeployClusterEnd: end}
+	if err := r.publishEvent(ctx, event); err != nil {
+		return fmt.Errorf("failed to report DeployClusterEnd event with ID %s: %w", event.GetUuid(), err)
+	}
+	log.V(1).Infof("Reported DeployClusterEnd event with ID %s", event.GetUuid())
+	return nil
+}
+
+// ReportCreateTopologyStart reports that a topology creation has started. The UUID of the
+// reported event is returned along with an error if the event reporting failed.
+func (r *Reporter) ReportCreateTopologyStart(ctx context.Context, t *epb.Topology) (string, error) {
+	event := newEvent()
+	event.Event = &epb.KNEEvent_CreateTopologyStart{CreateTopologyStart: &epb.CreateTopologyStart{Topology: t}}
+	if err := r.publishEvent(ctx, event); err != nil {
+		return "", fmt.Errorf("failed to report CreateTopologyStart event with ID %s: %w", event.GetUuid(), err)
+	}
+	log.V(1).Infof("Reported CreateTopologyStart event with ID %s", event.GetUuid())
+	return event.GetUuid(), nil
+}
+
+// ReportCreateTopologyEnd reports that a topology creation has ended.
```

Comment:
nit: unlike above you don't include the comment about errors in this description.

Reply:
done